src/etl/ProbingETLSource.cpp

```diff
@@ -0,0 +1,187 @@
+#include <etl/ProbingETLSource.h>
+
+ProbingETLSource::ProbingETLSource(
+    boost::json::object const& config,
+    boost::asio::io_context& ioc,
+    std::shared_ptr<BackendInterface> backend,
+    std::shared_ptr<SubscriptionManager> subscriptions,
+    std::shared_ptr<NetworkValidatedLedgers> nwvl,
+    ETLLoadBalancer& balancer,
+    boost::asio::ssl::context sslCtx)
+    : ioc_{ioc}
+    , sslCtx_{std::move(sslCtx)}
+    , sslSrc_{make_shared<SslETLSource>(
+          config,
+          ioc,
+          std::reference_wrapper{sslCtx_},
+          backend,
+          subscriptions,
+          nwvl,
+          balancer,
+          make_SSLHooks(mtx_))}
+    , plainSrc_{make_shared<PlainETLSource>(
+          config,
+          ioc,
+          backend,
+          subscriptions,
+          nwvl,
+          balancer,
+          make_PlainHooks(mtx_))}
```

Comment:
Do we need to pass this `mtx_` as a parameter?

`make_SSLHooks` and `make_plainHooks` are private members of `ProbingETLSource`, so they should have access to `mtx_`.

Is there another place where we construct the hooks w/ a different mutex?